src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -6,20 +6,34 @@
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final List<E> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
         // write your code here
+        try {
+            lock.writeLock().lock();
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
```

Comment:
It's a good practice to validate the 'index' parameter to ensure it's within the bounds of the list before attempting to access it. This can prevent IndexOutOfBoundsException.